File: scripts/screenshot.php
<?php
/**
 * Saves preview.png (1280x800) for every demo.html using headless Chrome.
 * Run from the repo root:  php scripts/screenshot.php
 * Set CHROME=/path/to/chrome if Chrome is not on your PATH.
 */
$root = dirname(__DIR__);
$chrome = getenv('CHROME') ?: trim(shell_exec('command -v google-chrome || command -v chromium || command -v chromium-browser'));
if (!$chrome) { fwrite(STDERR, "Chrome not found. Install Chrome or set CHROME.\n"); exit(1); }

foreach (glob("$root/*/*/demo.html") as $demo) {
    $png = dirname($demo) . '/preview.png';
    $cmd = sprintf('%s --headless=new --no-sandbox --disable-gpu --hide-scrollbars --window-size=1280,800 --virtual-time-budget=3000 --screenshot=%s %s 2>/dev/null',
        escapeshellarg($chrome), escapeshellarg($png), escapeshellarg('file://' . $demo));
    exec($cmd, $o, $rc);
    echo ($rc === 0 ? 'saved ' : 'FAILED ') . str_replace("$root/", '', $png) . "\n";
}

File: scripts/screenshot.py
"""Saves preview.png for every demo.html. Run: pip install playwright && playwright install chromium && python scripts/screenshot.py"""
import asyncio, glob, os
from playwright.async_api import async_playwright
ROOT = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
async def main():
    async with async_playwright() as p:
        b = await p.chromium.launch()
        pg = await b.new_page(viewport={"width": 1280, "height": 800})
        for f in sorted(glob.glob(os.path.join(ROOT, "*", "*", "demo.html"))):
            await pg.goto("file://" + f); await pg.wait_for_timeout(1200)
            await pg.screenshot(path=os.path.join(os.path.dirname(f), "preview.png")); print("saved", f)
        await b.close()
asyncio.run(main())
